### PR TITLE
[DOCS] Add Deeplink documentation

### LINKED ISSUE #497: Deep linking format
Referencing: https://github.com/RocketChat/Rocket.Chat.iOS/issues/440

We need to discuss and document our deep linking format so we can be consistent across all platforms.

Also, discuss which parameters to support so we can have nice deep-link features 🙂

### PR DESCRIPTION
**Pull request by rafaelks**
Closes #497

### CODE REVIEW

File: developer-guides/deeplink/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+> rocketchat://auth?host=foo.bar.com
+
+### With token
+
+**Params:**
+
+- host: The host of the server;
+- token: The token of the user to be authenticated;
+- userId: The id of the user to be authenticated;
```
Comment (cardoso) [inline developer-guides/deeplink/README.md]: Mark token and userId as optional. If not provided, we connect to the server, but show login form/methods.

**Review by MartinSchoeler — CHANGES_REQUESTED** [review]
Please add this file to the Table of Contents and Documentation Map.

File: developer-guides/deeplink/README.md (hunk 1 above)
Comment (rafaelks) [reply developer-guides/deeplink/README.md]: @cardoso I added two options (with token and without token) but let's make only one, simpler :-)

**Comment by rafaelks** [comment]
@MartinSchoeler How do I do that?

**Comment by MartinSchoeler** [comment]
@rafaelks Just add the file to its respective place in this file https://github.com/RocketChat/docs/blob/master/_data/toc.yml and in this one https://github.com/RocketChat/docs/tree/master/contributing/documentation/documentation-map (just add the file, no need for colored text on this one). You can find more info about the docs map and table of contents in this file https://github.com/RocketChat/docs/tree/master/contributing/documentation/contribution-guidelines

**Review by MartinSchoeler — APPROVED** [review]
Looks good to me!

**Pull request merged**